### PR TITLE
[FIX] account: fix write-after-post in PR 91638

### PR DESCRIPTION
**Pull request by antoine162**
#### The issue
PR #91638 introduced a write() on the partner_id  after the AML state
was set to 'posted'.

This causes a problem for users who have the journal hash activated.

#### The fix
Move the write() before posting the AMLs.

#### Related support tickets
opw-2858814, opw-2859214, opw-2859153

### CODE REVIEW

**Review by william-andre — APPROVED** [review]
@robodoo r+

**Comment by antoine162** [comment]
@robodoo r-

**Comment by qdp-odoo** [comment]
@robodoo r+ 
@antoine162 well fixed

File: addons/account/models/account_move.py [hunk 1]
```diff
@@ -2997,18 +2997,21 @@ def _post(self, soft=True):
...
-            'state': 'posted',
-            'posted_before': True,
-        })
 
         for move in to_post:
             # Fix inconsistencies that may occure if the OCR has been editing the invoice at the same time of a user. We force the
             # partner on the lines to be the same as the one on the move, because that's the only one the user can see/edit.
-            wrong_lines = move.is_invoice() and move.line_ids.filtered(lambda aml: aml.partner_id != move.commercial_partner_id)
+            wrong_lines = move.is_invoice() and move.line_ids.filtered(lambda aml: aml.partner_id != move.commercial_partner_id
+                                                                                   and aml.display_type not in ['line_section', 'line_note'])
```
Comment (jco-odoo) [inline addons/account/models/account_move.py:3005]: and not aml.display_type
```suggestion
                                                                                   and not aml.display_type
```

**Comment by antoine162** [comment]
@qdp-odoo Merciiii :blush:

File: addons/account/models/account_move.py (hunk 1 above)
Comment (qdp-odoo) [reply addons/account/models/account_move.py:3005]: ha yes, it's even better :sweat_smile:

Comment (antoine162) [reply addons/account/models/account_move.py:3005]: Thanks @jco-odoo

Comment (william-andre) [reply addons/account/models/account_move.py]: I'm gonna put it back in master anyways :shrug:

**Comment by antoine162** [comment]
@robodoo r+

**Comment by jco-odoo** [comment]
@robodoo r+

**Pull request closed without merging**